### ISSUE TITLE
Is it possible to have logging messages print to console without live logging

### ISSUE THREAD

**Issue by ArcLightSlavik** [pytest-dev/pytest#7704]
```
import logging


def some_function(number):
    print("i am a simple print")
    logging.info("i'm doing some stuff")
    logging.error("i'm in danger")
    return number * 2
```

```
from main import some_function


def test_some_function():
    number = 2
    after_number = some_function(2)
    assert after_number != number
    assert after_number == number * 2
```

If i run `pytest main_test.py -v -s` the `print` is displayed in the console, all good.
and if i do `log_cli = true` in `pytest.ini` `logging.error` is also displayed, all good.
But is there a way to have the `logging.error` display in the same way as the `print` without the `live log call` and so on? 
Even if it requires changes in pytest.ini.

**Comment by thisch** [comment]
No this is currently not possible. Can you provide a bit of context and explain why you want this feature?

**Comment by ArcLightSlavik** [comment]
If your running gunicorn your code might look like this:
```
import logging

from fastapi import FastAPI

logging.basicConfig(level=logging.INFO)

app = FastAPI()


@app.get('/hello')
def hello():
    print('hello')
    logging.info('a_logging_info')
    logging.error('a_logging_error')
    return 'HELLO, WORLD'
```
And if you run this locally with `gunicorn -k uvicorn.workers.UvicornWorker --bind 0.0.0.0:8088 app:app`
The logging will be like this:
```
hello
INFO:root:a_logging_info
ERROR:root:a_logging_error
```

Making a test to it:
```
from fastapi.testclient import TestClient
from app import app

client = TestClient(app)


def test_some_function():
    response = client.get('/hello')
    assert response.json() == dict(msg='Hello')
```
```
# pytest.ini
[pytest]
log_cli=true
log_level=INFO
```
Will result in this:
```
main_test.py::test_some_function hello

----------------------------------------------------------------------------------------------- live log call -----------------------------------------------------------------------------------------------
INFO     root:app.py:13 a_logging_info
ERROR    root:app.py:14 a_logging_error
PASSED
```
Which doesn't look bad but even here you can see that the `print` comes before the live log print.
Another example where a `print` comes before any info and then another `print('yoyo)` is now inside the live log call.
```
some_test.py::test_something_import yeye

----------------------------------------------------------------------------------------------- live log call -----------------------------------------------------------------------------------------------
INFO     monitor:some_file.py:157 An error happened during execution.
yoyo
INFO     monitor:some_test.py:84 ye
PASSED
```

TLDR: make an option for print/logging look closer to how gunicorn does it.

**Comment by thisch** [comment]
> Which doesn't look bad but even here you can see that the print comes before the live log print.

Yes, this is not well supported and we don't plan to change that, because we think that code-under-test should either use the logging infrastructure or use print statements and not use both. Therefore, we won't spent time on improving this.

If someone disagrees, feel free to reopen this.
